task-manager-diceroller/src/App.tsx

```diff
@@ -0,0 +1,167 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import "./App.css";
+import React from "react";
+import { TinyliciousClient } from "@fluidframework/tinylicious-client";
+import { ContainerSchema, IFluidContainer, ISharedMap, SharedMap } from "fluid-framework";
+import { ITaskManager, TaskManager } from "@fluid-experimental/task-manager";
+
+const client = new TinyliciousClient();
+
+const diceKey = "dice-key";
+const diceRollTaskId = "diceRollTaskId";
+
+const containerSchema: ContainerSchema = {
+	initialObjects: {
+		sharedMap: SharedMap,
+		taskManager: TaskManager,
+	},
+};
+
+const getInitialObjects = async () => {
+	let container: IFluidContainer;
+	const containerId = window.location.hash.substring(1);
+	if (!containerId) {
+		({ container } = await client.createContainer(containerSchema));
+		const id = await container.attach();
+		window.location.hash = id;
+	} else {
+		({ container } = await client.getContainer(containerId, containerSchema));
+	}
+	return {
+		sharedMap: container.initialObjects.sharedMap as ISharedMap,
+		taskManager: container.initialObjects.taskManager as ITaskManager,
+	};
+};
+
+function App() {
+	const [sharedMap, setSharedMap] = React.useState<ISharedMap>();
+	const [taskManager, setTaskManager] = React.useState<ITaskManager>();
+	const [diceValue, setDiceValue] = React.useState<number>();
+	const [assigned, setAssigned] = React.useState<boolean>(false);
+	const [queued, setQueued] = React.useState<boolean>(false);
+	const [subscribed, setSubscribed] = React.useState<boolean>(false);
+
+	React.useEffect(() => {
+		getInitialObjects().then((initialObjects) => {
+			setSharedMap(initialObjects.sharedMap);
+			setTaskManager(initialObjects.taskManager);
+		});
+	}, []);
+
+	React.useEffect(() => {
+		if (sharedMap !== undefined && taskManager !== undefined) {
+			// Update dice value from SharedMap
+			const syncView = () => setDiceValue(sharedMap.get(diceKey) ?? 1);
+			syncView();
+			sharedMap.on("valueChanged", syncView);
+
+			// TaskManager Setup
+			let rollInterval: NodeJS.Timeout | undefined;
```

Comment:
I think this is technically more correct (so it has correct typings in both Node and the browser):

```suggestion
			let rollInterval: ReturnType<typeof setInterval> | undefined;
```